l10n_ar_txt_tucuman/models/account_journal.py

```diff
@@ -0,0 +1,145 @@
+                x.l10n_latam_document_type_id.code
+            )['point_of_sale'] > 9999  # Verificar si el punto de venta es mayor a 9999
+        )
+        if move_lines_with_five_digits_pos:
+            raise UserError(_("Algunos comprobantes tienen punto de venta de 5 dígitos y deben tener de 4 dígitos para "
+                              "poder generar el archivo txt de retenciones y percepciones de Tucuman: %s") %
+                            (", ".join(move_lines_with_five_digits_pos.mapped('move_id.name'))))
+        percepciones = move_lines.filtered(lambda x: x.l10n_latam_document_type_id.internal_type == 'invoice')
+        if percepciones and len(percepciones) != len(move_lines):
+            raise UserError(_("Debe generar archivos para TXT Tucuman por separado para retenciones por un lado y "
+                              "percepciones por otro."))
+        return percepciones if percepciones else self.env['account.move.line']
+
+    def _iibb_tucuman_datos_txt_file(self, line, percepciones):
+        """ Archivo DATOS.TXT para Tucuman. """
+        content_datos = ''
+        # 1, FECHA, longitud: 8. Formato AAAAMMDD
+        content_datos += fields.Date.from_string(line.date).strftime('%Y%m%d')
+        # 2, TIPODOC, longitud: 2
+        content_datos += line.partner_id.l10n_latam_identification_type_id.l10n_ar_afip_code
+        # 3, DOCUMENTO, longitud: 11
+        content_datos += line.partner_id.l10n_ar_vat
+        # 4, TIPO COMP, longitud: 2
+        # 99 para retenciones por el ejemplo que pasó en el archivo adjunto el cliente en la tarea 38200
+        content_datos += line.move_id.l10n_latam_document_type_id.code.zfill(2) if line in percepciones else '99'
+        # 5, LETRA, longitud: 1
+        content_datos += line.move_id.l10n_latam_document_type_id.l10n_ar_letter if line in percepciones else ' '
+        # 6, COD. LUGAR EMISION, longitud: 4
+        document_number_parts = line.move_id._l10n_ar_get_document_number_parts(
+            line.move_id.l10n_latam_document_number, line.l10n_latam_document_type_id.code)
+        content_datos += str(document_number_parts['point_of_sale']).zfill(4)
+        # 7, NUMERO, longitud: 8
+        content_datos += str(document_number_parts['invoice_number']).zfill(8)
+        # 8, BASE_CALCULO, longitud: 15,2
+        content_datos += '%015.2f' % (line.tax_base_amount if percepciones else line.payment_id.withholding_base_amount)
+        # 9, PORCENTAJE/ALICUOTA, longitud: 6,3
+        partner_alicuot = line.tax_line_id.get_partner_alicuot(line.partner_id, line.date)
+        if line in percepciones:
+            content_datos += '%06.3f' % partner_alicuot.alicuota_percepcion
+        else:
+            content_datos += '%06.3f' % partner_alicuot.alicuota_retencion
+        # 10, MONTO_RET/PER, longitud: 15,2
+        content_datos += '%015.2f' % abs(line.balance)
+        content_datos += '\r\n'
+        return content_datos if content_datos else ''
```

Comment:
you already initialize the content_ncfact = ''
```suggestion
        return content_datos
```